stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -243,15 +246,24 @@ internal constructor(
         }
     }
 
+    internal fun launch(
+        block: suspend CoroutineScope.() -> Unit
+    ) = userScope.launch(block = block)
+
+    @InternalStreamChatApi
+    public fun inheritScope(block: (Job) -> CoroutineContext): CoroutineScope {
+        return userScope + block(userScope.coroutineContext.job)
+    }
```

Comment:
Can we add docs to this method?
When should we use it?
Why use it and not the `userScope` directly?